### PR TITLE
feat: UI 2.0 Avatar

### PR DESCRIPTION
**Pull request by calebpanza**
## Motivation
Next on the list for components is the Avatar component. The goal, as always, was to simplify the implementation and make it look/function nearly 1-1 like the Figma counterpart.

## Considerations
We don't have a `ButtonIcon` component, so I've withheld the `icon` and `onPress` properties until that component gets created. That will be a future PR.

You'll notice the introduction of "t-shirt sizes" in the `size` property. This is due to this component only using a specific set of sizing according to Figma, and so there's no need to expose extra functionality like full size control.

## Testing
Test coverage is at 100%. Stories are previewed below.

| Basic | Placeholder | Is Active | Emoji |
| --- | --- | --- | --- |
| Simulator Screen Shot - iPad Air (5th generation) - 2022-09-15 at 16 51 46 | Simulator Screen Shot - iPad Air (5th generation) - 2022-09-15 at 16 51 51 | Simulator Screen Shot - iPad Air (5th generation) - 2022-09-15 at 16 51 56 | Simulator Screen Shot - iPad Air (5th generation) - 2022-09-15 at 16 52 05 |

### CODE REVIEW

File: templates/mobile/src/ui-kit/components/Avatar/Avatar.js [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+  const [shouldShowInitials, setShouldShowInitials] = useState(true);
+  const _i = initials(profile?.firstName, profile?.lastName);
+
+  return (
+    <Styled>
+      <AvatarImage
+        size={size}
+        source={profile?.photo}
+        initials={_i}
+        onError={() => setShouldShowInitials(true)}
```
Comment (vinnyjth) [inline templates/mobile/src/ui-kit/components/Avatar/Avatar.js:29]: good idea!

File: templates/mobile/src/ui-kit/components/Avatar/utils.js [hunk 2]
```diff
@@ -0,0 +1,22 @@
+import compact from 'lodash/compact';
+import Color from 'color';
+
+export const initialsToColor = (initials, s = 40, l = 60) => {
```
Comment (vinnyjth) [inline templates/mobile/src/ui-kit/components/Avatar/utils.js:4]: niiiiice

File: templates/mobile/src/ui-kit/components/Avatar/Avatar.styles.js [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+}))(({ theme }) => ({
+  position: 'absolute',
+  height: '100%',
+  width: '100%',
+  display: 'flex',
+  alignItems: 'center',
+  justifyContent: 'center',
+  backgroundColor: Color(theme.colors.fill.paper).fade(0.4).hexa(),
+  zIndex: 1000,
+  borderRadius: 9999,
```
[the remark this answers is not in the document]
Comment (calebpanza) [reply templates/mobile/src/ui-kit/components/Avatar/Avatar.styles.js:82]: Breaking out each "element" into its own component worked out real nice, I think! I feel like it's pretty readable. Lol

**Review by vinnyjth — APPROVED** [review]

File: templates/mobile/src/ui-kit/components/Avatar/utils.js (hunk 2 above)
Comment (calebpanza) [reply templates/mobile/src/ui-kit/components/Avatar/utils.js:4]: This is all @conrad-vanl . Lol. I just ported it

**Review by tristanguckenberger — APPROVED** [review]

**Pull request merged**